pkg/body-based-routing/handlers/request.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+	"encoding/json"
+
+	basepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	eppb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+)
+
+// HandleRequestBody handles request bodies.
+func (s *Server) HandleRequestBody(body *eppb.HttpBody) (*eppb.ProcessingResponse, error) {
+	var data map[string]any
+	if err := json.Unmarshal(body.GetBody(), &data); err != nil {
+		return nil, err
+	}
+
+	modelVal, ok := data["model"]
+	if !ok {
+		return &eppb.ProcessingResponse{
+			Response: &eppb.ProcessingResponse_RequestBody{
+				RequestBody: &eppb.BodyResponse{},
+			},
+		}, nil
+	}
+
+	modelStr, ok := modelVal.(string)
+	if !ok {
+		return &eppb.ProcessingResponse{
+			Response: &eppb.ProcessingResponse_RequestBody{
+				RequestBody: &eppb.BodyResponse{},
+			},
+		}, fmt.Errorf("the model parameter value %v is not a string", modelVal)
+	}
+
+	return &eppb.ProcessingResponse{
+		Response: &eppb.ProcessingResponse_RequestBody{
+			RequestBody: &eppb.BodyResponse{
+				Response: &eppb.CommonResponse{
+					// Necessary so that the new headers are used in the routing decision.
+					ClearRouteCache: true,
+					HeaderMutation: &eppb.HeaderMutation{
+						SetHeaders: []*basepb.HeaderValueOption{
+							{
+								Header: &basepb.HeaderValue{
+									Key:   "model",
+									Value: modelStr,
```

Comment:
Although there's no standard defined max length for HTTP header values, it does seem like it could be useful to add some kind of truncation here at some large number of characters so we're not propagating completely unbounded values. Maybe 1000 chars would work?

/cc @ahg-g @andresguedez @danehans

Reply:
Do you think it makes sense to file an issue for it and take it in a follow-up PR?